atArgumentParser.go

```diff
@@ -2,102 +2,171 @@ package vmcommon
 
 import (
 	"encoding/hex"
-	"strings"
 )
 
-type atArgumentParser struct {
+// AtArgumentParser is a parser that splits arguments by @ character
```

Comment:
Why did is this public? It implements an interface, any access should be through that.

Reply:
Forcing client code to use such as simple abstraction only by means of an interface is bad design. The **VM arguments parser** can be, and it is, indeed, used via an interface most of the time in the project, but that should not be a necessity.

**One should depend upon abstractions, not concretions.** - that is true, of course. The parser itself does not cross "architecturally significant boundaries" (does not read files, does not query external services) in order to achieve its purpose, and actually, does not have any other dependency - quite enclosed. Similar to how one does not need to define (in day-to-day programming) an interface for an Integer - since the Integer is abstracted enough, one does not necessarily need to define an interface over this simple abstraction.

Also, we have to be reasonable and pragmatic programmers: it helped nothing to define interfaces for this dummy parser all over the project (2 places Node, 1 place Arwen). Mocking it server little purpose - setting up errors to be raised by the parser would have been more trivial and explicit by simply setting error triggering input - instead of faking error conditions.

Reply:
Update `AtArgumentParser` in comment.

Reply:
Fixed.